ctest-next/src/lib.rs

```diff
@@ -54,16 +57,17 @@ pub enum VolatileItemKind {
 pub(crate) enum MapInput<'a> {
     /// This variant is used for renaming the struct identifier.
     Struct(&'a Struct),
+    Union(&'a Union),
     Fn(&'a crate::Fn),
-    #[expect(unused)]
-    Field(&'a Struct, &'a Field),
+    Field(Either<&'a Struct, &'a Union>, &'a Field),
     Alias(&'a Type),
     Const(&'a Const),
     Static(&'a Static),
     Type(&'a str),
     /// This variant is used for renaming the struct type.
     StructType(&'a str),
     UnionType(&'a str),
+    FieldType(Either<&'a Struct, &'a Union>, &'a Field),
```

Comment:
Could these be split to `StructField` `StructFieldType` `UnionField` and `UnionFieldType` rather than using `Either`? Seems slightly more straightforward since you have to `match` somewhere anyway.